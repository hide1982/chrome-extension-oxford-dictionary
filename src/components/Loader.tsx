import styled from "styled-components"

const Loader = styled.div.attrs({ "data-testid": "loader" })<{
  isShow: boolean
}>`
  &,
  &::before,
  &::after {
    pointer-events: none;
    background: #000;
    animation: load1 1s infinite ease-in-out;
    width: 1em;
    height: 4em;
  }

  &::before,
  &::after {
    position: absolute;
    top: 0;
    content: "";
  }

  & {
    display: ${({ isShow }) => (isShow ? "block" : "none")};
    color: #000;
    position: relative;
    font-size: 9px;
    animation-delay: -160ms;
  }

  &::before {
    left: -1.5em;
    animation-delay: -320ms;
  }

  &::after {
    left: 1.5em;
  }

  @keyframes load1 {
    0%,
    80%,
    100% {
      box-shadow: 0 0;
      height: 4em;
    }
    40% {
      box-shadow: 0 -2em;
      height: 5em;
    }
  }
`

export default Loader
