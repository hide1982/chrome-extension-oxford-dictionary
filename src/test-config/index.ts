import "@testing-library/jest-dom"

import "@/test-config/matchMedia"
