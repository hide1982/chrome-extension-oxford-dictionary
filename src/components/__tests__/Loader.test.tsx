import React from "react"
import "@/test-config"
import { render, screen } from "@testing-library/react"

import Loader from "@/components/Loader"

describe("Loader component", () => {
  test("snapshot visible", () => {
    render(<Loader isShow={true} />)

    expect(screen.getByTestId("loader")).toMatchSnapshot()
  })

  test("snapshot invisible", () => {
    render(<Loader isShow={false} />)

    expect(screen.getByTestId("loader")).toMatchSnapshot()
  })
})
