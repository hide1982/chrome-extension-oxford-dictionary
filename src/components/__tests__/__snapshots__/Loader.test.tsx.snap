// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component snapshot invisible 1`] = `
<div
  class="sc-bdfBwQ gUbHnz"
  data-testid="loader"
/>
`;

exports[`Loader component snapshot visible 1`] = `
<div
  class="sc-bdfBwQ jPdVrk"
  data-testid="loader"
/>
`;
